Libraries/TestHelpers.psm1

```diff
@@ -721,6 +721,40 @@ Function Get-LISAv2Tools($XMLSecretFile) {
 	}
 }
 
+Function Get-SSHKey ($XMLSecretFile) {
+	# Download SSHKey when provide a URL
+	$temp_Folder = $env:TEMP
+	$sshKeyPath = [string]::Empty
+	if ($XMLSecretFile) {
+		$WebClient = New-Object System.Net.WebClient
+		$xmlSecret = [xml](Get-Content $XMLSecretFile)
+		$privateSSHKey = $xmlSecret.secrets.sshPrivateKey.InnerText
+		if ($privateSSHKey) {
+			$sshKeyPath = $privateSSHKey
+		}
+	}
+	if ($privateSSHKey) {
+		if ($privateSSHKey -match "^(http|https)://") {
+			$WebClient = New-Object System.Net.WebClient
+			$privateSSHKeyName = $privateSSHKey.Split('?')[0].Split('/')[-1]
+			try {
+				$WebClient.DownloadFile("$privateSSHKey", "$temp_Folder/$privateSSHKeyName")
+			} catch {
+				Throw "Failed to download from $privateSSHKey, please double check the path."
+			}
+			$sshKeyPath = "$temp_Folder/$privateSSHKeyName"
+		}
+		if (![System.IO.File]::Exists($sshKeyPath)) {
```

Comment:
remember to clean up and force delete those ppk  files from AzureProvider -> [void] RunTestCleanup()

Reply:
Do we delete the tempfolder after testing?

Reply:
either, we can delete the folder, or just delete the *.ppk file.   for now there are deploy templates in that folder, maybe.   So, it want to keep those files for investigation, just delete .ppk files.